From 1cf00485ff6fb41ea6a8cc0a2b29804c6c1181d8 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Fri, 4 Nov 2011 10:05:45 +0800
Subject: [PATCH 201/498] ENGR00161379 fix dual display problem.

when video combined with UI to hdmi, then video playblack
finishes, UI can not show in hdmi.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx5x/hwcomposer/BG_device.cpp  | 10 ++++++++--
 mx5x/hwcomposer/hwcomposer.cpp |  7 ++++++-
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/mx5x/hwcomposer/BG_device.cpp b/mx5x/hwcomposer/BG_device.cpp
index 7826449..15fdd58 100755
--- a/mx5x/hwcomposer/BG_device.cpp
+++ b/mx5x/hwcomposer/BG_device.cpp
@@ -18,7 +18,7 @@
 #include <sys/ioctl.h>
 #include <hardware/hardware.h>
 //#include <hardware/overlay.h>
-
+#include <cutils/properties.h>
 #include <fcntl.h>
 #include <errno.h>
 
@@ -161,12 +161,18 @@ int BG_device::uninit()
 {
 	  //int status = -EINVAL;    
     int blank = 1;
+    char value[10];
     HWCOMPOSER_LOG_RUNTIME("---------------BG_device::uninit()------------");
 
     if(ioctl(m_dev, FBIOBLANK, blank) < 0) {
 	    HWCOMPOSER_LOG_ERR("Error!BG_device::uninit BLANK FB2 failed!\n");
         //return -1;
-    }	  
+    }
+    property_get("media.VIDEO_PLAYING", value, "0");
+    if (strcmp(value, "0") == 0) {
+        blank = 0;
+        ioctl(m_dev, FBIOBLANK, blank);
+    }
     munmap((mbuffers[0]).virt_addr, (mbuffers[0]).size * DEFAULT_BUFFERS);
     close(m_dev);
 
diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index 8db2924..de2e673 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -80,7 +80,7 @@ static void dump_layer(hwc_layer_t const* l) {
             l->displayFrame.right,
             l->displayFrame.bottom);
 }
-
+static int releaseAllOutput(struct hwc_context_t *ctx);
 static int hwc_check_property(hwc_context_t *dev)
 {
     //bool bValue = false;
@@ -102,6 +102,11 @@ static int hwc_check_property(hwc_context_t *dev)
        if (strcmp(value, "1") == 0){
            property_set("sys.VIDEO_OVERLAY_DISPLAY", "0");
            property_set("sys.VIDEO_DISPLAY", "1");
+
+           property_get("media.VIDEO_PLAYING", value, "0");
+           if (strcmp(value, "0") == 0) {
+               releaseAllOutput(dev);
+           }
        }
        else{
            property_set("sys.VIDEO_OVERLAY_DISPLAY", "2");
-- 
1.8.0

